[Trust] Added Monberaux's skills

thecursedcometh2 opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Adds Monberaux's item skills (Except Final Elixir and Gold Needle)
- Adds proper effects to mix_guard_drink skill
- It should be noted that the skill names in the combat log are wrong, but the effects and animations remain accurate. 
- Also it should be noted that this is my first proposed changes on this project, or on GitHub in general. I followed the contribution guide to the best of my ability, but please let me know if I have done anything wrong. I intend to contribute a lot more to LSB and I want to make it as smooth as possible for everyone involved.

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
Summon Trust: Monberaux and observe his new behaviors and their effects
<!-- Clear and detailed steps to test your changes here -->

thecursedcometh2 committed:
Added Monberaux skills

thecursedcometh2 committed:
Fix mix_guard_drink

thecursedcometh2 commented:
I noticed that the sanity check failed because there were empty lines in the file. Do I need to remove those? I saw other files with empty lines so I thought it was alright.

thecursedcometh2 committed:
Fixed Monberaux.lua format

Xaver-DaRed commented:
First and foremost, welcome.
So...
- You are using tabs. Replace them with 4 spaces.
- You can have no more than one empty line in between code. so...
```lua
This is

ok 
```
```lua
This is


not ok
```

thecursedcometh2 commented:
> First and foremost, welcome. So...
> 
>     * You are using tabs. Replace them with 4 spaces.
> 
>     * You can have no more than one empty line in between code. so...
> 
> 
> ```lua
> This is
> 
> ok 
> ```
> 
> ```lua
> This is
> 
> 
> not ok
> ```

Thanks for letting me know. I changed the file based on the complaints that the sanity checker gave me, removing the extra lines and replacing tabs with 4x spaces. Is the new file improved?

zach2good commented on scripts/globals/mobskills/mix_guard_drink.lua:
```diff
@@ -8,6 +8,9 @@ mobskillObject.onMobSkillCheck = function(target, mob, skill)
 end
 
 mobskillObject.onMobWeaponSkill = function(target, mob, skill)
+    target:addStatusEffect(xi.effect.PROTECT, 220, 0, 300)
+    target:addStatusEffect(xi.effect.SHELL, 2930, 0, 300)
```
https://ffxiclopedia.fandom.com/wiki/Trust:_Monberaux
https://www.bg-wiki.com/ffxi/Monberaux

The wikis agree on 220 for the Protect value, but where did you get the Shell values for this from? Did you get this from retail? Protect 220 vs Shell 2930 doesn't seem right (though, I don't really know much about the values of these sorts of thing)

Does this scale with level?

TeoTwawki replied:
@zach2good wiki's show a x/256ths scale for shell where we use a larger basis to do a percentage down to multiple decimal points, hence the discrepancy. So what he has here is 29.30%

thecursedcometh2 replied:
Teo is right, 2930 for Shell is 29.30%. I referenced the values for Protect and Shell V, which BGWiki says is the potency of the guard drink move.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
     -- Tends to be particular about which potions to use. Seems to favor healing for just the
     -- right amount of HP instead of defaulting to the highest-rank potion.
-    -- mob:addSimpleGambit(ai.t.PARTY, ai.c.HP_MISSING, 500, ai.r.MS, ai.s.SPECIFIC, 4236, healingMoveCooldown) -- Max Potion
+    mob:addMod(xi.mod.MPP, -80)
+    mob:addMod(xi.mod.HPP, 50)
```
Where are these values from?

thecursedcometh2 replied:
The -Max MP is from BGWiki, which says he has -80% MP. The wiki also says his stats and health scale as if he was a tank job, and yet he had very low health from my testing on my own server (~1800 for paladin elvaan). I couldn't figure out a way to edit his stats normally so I gave him an HP modifier. I'll get rid of it if it's not acceptable, as I wasn't sure how to make him not so squishy.

TeoTwawki replied:
fun fact: he basically IS an PLD elvaan, even has the traits.

Our stat basis **should** be coming from the rankings in mob family system but right now I don't know if that is really the case (I recall it being bs made up in core instead at one point) **and** I think all or almost all trusts are using the same generic humanoid for their family ID (145 iirc) when we probably need several (especially for the non humanoid ones like the Goobue).

TeoTwawki replied:
I would consider the mp one legit, for whatever reason SE decided to nerf his native pld mp pool even tho he has no spells.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
-    -- mob:addSimpleGambit(ai.t.PARTY, ai.c.HP_MISSING, 500, ai.r.MS, ai.s.SPECIFIC, 4236, healingMoveCooldown) -- Max Potion
+    mob:addMod(xi.mod.MPP, -80)
+    mob:addMod(xi.mod.HPP, 50)
+    mob:addSimpleGambit(ai.t.CASTER, ai.c.MPP_LT, 25, ai.r.MS, ai.s.SPECIFIC, 4254, healingMoveCooldown)
```
No comment here, what move is this?

thecursedcometh2 replied:
Skill 4254 is Mix: Dry Ether Concoction, which he uses to restore MP on retail for himself and others. I added the caster condition because otherwise he kept spamming it on jobs with no MP.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.PARALYSIS, ai.r.MS, ai.s.SPECIFIC, 4247, healingMoveCooldown) -- Mix: Para-B-Gone
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS_FLAG, xi.effectFlag.ERASABLE, ai.r.MS, ai.s.SPECIFIC, 4245, healingMoveCooldown) -- Mix: Panacea-1
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.REGEN, ai.r.MS, ai.s.SPECIFIC, 4257, healingMoveCooldown) -- Mix: Life Water
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.STR_BOOST, ai.r.MS, ai.s.SPECIFIC, 4261, healingMoveCooldown)
```
No comment here, what move is this?

thecursedcometh2 replied:
Skill 4261 is Mix: Samson's Strength, which is an all attribute boost for 1 minute.

TeoTwawki replied:
@thecursedcometh2 you should just add the comment to the code.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.STR_BOOST, ai.r.MS, ai.s.SPECIFIC, 4261, healingMoveCooldown)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_DEF_BOOST, ai.r.MS, ai.s.SPECIFIC, 4259, healingMoveCooldown) -- Mix: Dragon Shield
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_ATK_BOOST, ai.r.MS, ai.s.SPECIFIC, 4258, healingMoveCooldown)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.PROTECT, ai.r.MS, ai.s.SPECIFIC, 4255, healingMoveCooldown)
```
No comment here, what move is this?

thecursedcometh2 replied:
Skill 4258 is Mix: Elemental Power, which is an AOE +20 MAB for the party.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_DEF_BOOST, ai.r.MS, ai.s.SPECIFIC, 4259, healingMoveCooldown) -- Mix: Dragon Shield
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_ATK_BOOST, ai.r.MS, ai.s.SPECIFIC, 4258, healingMoveCooldown)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.PROTECT, ai.r.MS, ai.s.SPECIFIC, 4255, healingMoveCooldown)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.NEGATE_SLEEP, ai.r.MS, ai.s.SPECIFIC, 4256, healingMoveCooldown)
```
No comment here, what move is this?

thecursedcometh2 replied:
Skill 4255 is Mix: Guard Drink, which acts as a 5 minute Protectra and Shellra 5. Skill 4256 is Insomniant, which is a Negate Sleep effect for Monberaux only. Which I'm now realizing was disabled on LSB. I enabled it on my server and it worked with no errors, but I forgot to push a commit to enable that. My bad.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
     xi.trust.message(mob, xi.trust.message_offset.SPAWN)
 
-    -- local healingMoveCooldown = 10
+    local healingMoveCooldown = 3
```
This was a placeholder, did you test this delay on retail?

thecursedcometh2 replied:
I referenced BGWiki again for this, which says that he uses an ability every 3-4 seconds as needed.

TeoTwawki replied:
we should check in game, as that is someone going "its about such and such" without measuring or they wouldn't say "about" nor "3-4" they'd give an exact value.

thecursedcometh2 replied:
From my experience on retail, it looks to be 3 seconds on the dot. At least judging by the in game combat log. The discrepancy on the wiki could be because of high ping or something, I'm not sure.

TeoTwawki replied:
likely whatever multiple of the battle tick rate amounts to 3 so likely within mere milliseconds of 3.

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,30 @@ spellObject.onSpellCast = function(caster, target, spell)
 end
 
 spellObject.onMobSpawn = function(mob)
-    -- TODO: Summon: I am Doctor Monberaux. My services are available for any affliction.
```
👍

TeoTwawki commented:
> The trick here to get the messages to show up correctly is to get Monberaux to enter CItemState when using his abilities, rather than CMobSkillState - that will force him to send the correct packets. He's not actually faking using items in this regard, he's actually using them on the packet level.

I'd say its technically still "faking" since there is no actual item (I mean, he's got no inventory right?) but he needs the item type packets for it. this is another point in favor of having the action type as an available field instead of the skill list always assuming ws <256 and mobskill type > 255 and us always having to workaround it for these exceptions that keep popping up.

thecursedcometh2 commented:
I'm sorry for the lack of comments and sources on my changes, I'll be better prepared for that in the future.

thecursedcometh2 committed:
Added comments and removed HP modifier

thecursedcometh2 committed:
Enabled skill 4256, Insomniant for Monberaux

thecursedcometh2 commented:
Alright, I added the relevant comments and removed the HP modifier. Also I enabled Mix: Insomniant in the mob_skills.sql because I forgot that was disabled on LSB. It works though, no errors of any kind.

Xaver-DaRed commented:
If you have made the changes, the one that originally requested the changes has to remove that message.

However, one of us also has to test the PR locally aswell.

So It's a matter of waiting, pretty much.

thecursedcometh2 commented:
> If you have made the changes, the one that originally requested the changes has to remove that message.
> 
> However, one of us also has to test the PR locally aswell.
> 
> So It's a matter of waiting, pretty much.

Got it, thank you.

zach2good commented:
This block (while still wrong) makes the messaging less confusing:

```lua
    mob:addListener("WEAPONSKILL_USE", "MONBERAUX_WS", function(mobArg, targetArg, skillid, spentTP, action)
        action:setCategory(9) -- ACTIONTYPE::ACTION_ITEM_START
    end)
```

Before:
<img width="257" alt="image" src="https://github.com/LandSandBoat/server/assets/1389729/b287b935-1d60-47a2-9f93-1bba07f30c4a">

After:
<img width="233" alt="image" src="https://github.com/LandSandBoat/server/assets/1389729/27e63c4b-f52f-4724-a2ca-aef09020cdb8">

Rather than writing a whole new path in the mob_controller and item/mobskill states just so Moberaux can somewhat convincingly use items, it's probably a lot better to mess with the outgoing action (which generates the packet). Have a play around with that block and see if you can get the right result. There's plenty of caps in the caps discord to compare with.

thecursedcometh2 commented:
> 

I'll give it a go. By "play around", you mean change the category and see what happens?

thecursedcometh2 commented:
So when I have that block in his file, Monberaux no longer plays an animation when using an item. He just stands there and the skill effect happens. The skill name is #0 like in your screenshot. Did this happen for you, too?

thecursedcometh2 commented:
Ok so: setCategory(3) gives the right skill name, but removes his item using animation.

thecursedcometh2 commented:
Alright, I tried all 37 action types, and #3 is the only one that had the right names. Sadly it looks like, without a custom action type, we'll have to choose between the item animation and the correct skill name.

WinterSolstice8 committed:
[core] Add another bit to pack in action ID for mobskill finish

WinterSolstice8 committed:
[trust] Add proper messaging for some Monberaux abilities

* Most no effects are guessed
* Insomniant is disabled due to a bad animation and no cap

WinterSolstice8 commented:
I added two commits to your PR to fix messaging, action IDs and attempted proper messaging at the abilities I could find caps for.
Insomniant is disabled because the animation is wrong and I can't seem to find a cap for it.

WinterSolstice8 commented:
Note that messaging order is wrong, Monberaux targets the player first instead of himself like he should. That can get fixed in a future PR.

thecursedcometh2 committed:
Merge branch 'LandSandBoat:base' into TrustMonberaux

thecursedcometh2 commented:
So that was definitely an improvement! He uses the right animation now and the combat log says "Monberaux readies (skill here)". However, when he actually uses the skill, it always says "Monberaux uses Blade: Hi". Not sure what's up with that. Also he doesn't use Para-B-Gone for some reason. Will look into it.

thecursedcometh2 commented:
Ah well if the scripting for Para-B-Gone is wrong I'll just comment out the skill for now.

WinterSolstice8 commented:
![image](https://github.com/LandSandBoat/server/assets/60417494/4af3fa92-ee36-4cc4-8951-8d9217104b7d)
Seems to mostly work right for me. Not sure why the all status parameters isn't there but for my player but thats probably from artifacts because battlemod and monberaux being fairly new?

thecursedcometh2 commented:
Yeah I don't use battlemod so I get the Blade: Hi message instead. It still displays the right name when he's about to use it though so it's probably fine.

WinterSolstice8 commented:
![image](https://github.com/LandSandBoat/server/assets/60417494/de1707a3-a51f-43c6-b105-7448aefc0564)
works on my machine

thecursedcometh2 commented:
Hmm... I can't imagine what it would be on my end, but I'll do a fresh install and see if that fixes it.

Xaver-DaRed commented:
If I may ask, did you recompile your local server with WinterSolstice changes before testing them?

thecursedcometh2 commented:
> If I may ask, did you recompile your local server with WinterSolstice changes before testing them?

I did, that's why I was confused as to why the results were different.

thecursedcometh2 commented:
![Untitled](https://github.com/LandSandBoat/server/assets/136541954/10206986-b8bc-4d53-94a0-86ddab627b08)

Alright I completely reset my server but it's still giving me the Blade: Hi thing. I have no idea why it's right for you but wrong for me... but hey, if it works for other people, then that's good. I'll have to figure out what's happening to me at some point.

thecursedcometh2 commented:
![Idris_2023 06 17_202005](https://github.com/LandSandBoat/server/assets/136541954/757be0a5-d5c6-4a69-b87e-502bbef24262)
It's still happening for me... did a complete reinstall of the server. Dunno what's going on with me but if it works for you, then it works.

TeoTwawki commented:
There is no installation nor reinstallation of the server - we don't make or ship an installer. we offer source code and you build or rebuild. if you have the same database and the same code its going to be the same every time. So look for whats different between the branch(code) they are testing vs what you are testing at your computer. Something IS different, its physically impossible for it not to be. It would be like finding the speed of light in a vacuum different just for one person.

thecursedcometh2 commented:
> There is no installation nor reinstallation of the server - we don't make or ship an installer. we offer source code and you build or rebuild. if you have the same database and the same code its going to be the same every time. So look for whats different between the branch(code) they are testing vs what you are testing at your computer. Something IS different, its physically impossible for it not to be. It would be like finding the speed of light in a vacuum different just for one person.

Sorry, what I mean is that I deleted the server folder and rebuilt it from scratch from my branch. With Git and Visual Studio, like I had never installed it before. As for the files, there is no difference between my local files and the ones on the repository. I checked them before building, all 14 files are exactly the same line for line.

TeoTwawki commented:
>  what I mean is that I deleted the server folder and 

Yeah but see you'll just be redownloading exactly the same thing you had before you deleted it. It just throws away your time. Look for anything different - a merge commit something might have stealth changed in, an edit conflict that you had to resolve, a discrepancy between your actual running database vs what in you your `.sql` files. Something is there somewhere.

Xaver-DaRed commented:
I downloaded your branch and did a fresh build and I got the same result as WinterS.

Can I ask, how are you building the server? what method (guide or direct download)? what concrete build (linux, windows)?

This is important because, as you said, you want to help contributing. A difference this big suggest an error somewhere and finding it will help both you and us in the long run.

thecursedcometh2 commented:
> I downloaded your branch and did a fresh build and I got the same result as WinterS.
> 
> Can I ask, how are you building the server? what method (guide or direct download)? what concrete build (linux, windows)?
> 
> This is important because, as you said, you want to help contributing. A difference this big suggests an error somewhere and finding it will help both you and us in the long run.

I got it working, but I'll detail this for you in case I'm doing something wrong.

First I build the default LSB server via the quick start guide: https://github.com/LandSandBoat/server/wiki/Quick-Start-Guide

Then I download all of the files from this pull request and replace the files in my local server folder. Then I rebuild the database with the powershell command: py -3 dbtool.py

I did this just now and it fixed it. I believe what I did wrong last time was attempting to install my own fork instead of the default LSB repository. Something is likely wrong with my fork, but I'm not sure what. Once this is merged I'll make a new, fresh fork and stop using my current one.

TeoTwawki commented:
excellent that you figured it out, thank you for the update!

thecursedcometh2 commented:
Well I'm glad we got this working, Monberaux is one of my favorite trusts. Is the work all done, or does anything else need to be done before it can be merged?

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -32,13 +45,9 @@ spellObject.onMobSpawn = function(mob)
-    -- mob:addListener("WEAPONSKILL_USE", "MONBERAUX_ITEM_WEAPONSKILL_USE", function(mobArg, target, wsid, tp, action)
-    --     action:setCategory(5) -- ACTION_ITEM_FINISH
-    -- end)
+    mob:addListener("WEAPONSKILL_USE", "MONBERAUX_WS", function(mobArg, targetArg, skillid, spentTP, action)
```
This probably wants the above comment kept/adjusted to explain _why_ we need this

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,29 @@ spellObject.onSpellCast = function(caster, target, spell)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_DEF_BOOST, ai.r.MS, ai.s.SPECIFIC, 4259, healingMoveCooldown) -- Mix: Dragon Shield
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.MAGIC_ATK_BOOST, ai.r.MS, ai.s.SPECIFIC, 4258, healingMoveCooldown) -- Mix: Elemental Power
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.PROTECT, ai.r.MS, ai.s.SPECIFIC, 4255, healingMoveCooldown) -- Mix: Guard Drink
+    -- mob:addSimpleGambit(ai.t.PARTY, ai.c.NOT_STATUS, xi.effect.NEGATE_SLEEP, ai.r.MS, ai.s.SPECIFIC, 4256, healingMoveCooldown) -- Insomniant
```
Can we get a comment explaining why insomniant is disabled, and what would be needed to fix/enable it?

zach2good commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,29 @@ spellObject.onSpellCast = function(caster, target, spell)
 end
 
 spellObject.onMobSpawn = function(mob)
```
I have no doubt that downstream users are going to come in and mess with Monberaux for balance and whatever. Can you collect (and comment somewhere prominently) all the various TODOs in here and from his moves so it's clear what he needs to be completed/verified - if that's not going to happen in this PR

thecursedcometh2 committed:
Added todo's and adjusted effects

Added Paralyze resistance (retail has paralyze immunity) and adjusted MP% -80 to -90 (was wrong value).  Added list of todo's for downstream.

thecursedcometh2 commented:
Added the changes that Zach requested. I also added mob:addMod(xi.mod.PARALYZERES, 100) because on retail he is functionally immune to paralyze. However, I don't know for sure if 100 = paralyze immune. Could anyone verify this? Not sure where to find mod values.

thecursedcometh2 committed:
Added todos for other skills

Xaver-DaRed commented:
So...

PARALYZERES is the modifier used by the "Resist Paralyze" job trait.
PARALYZE:MEVA is the modifier used to give more evasion against paralyze.

The main difference is that when the job trait triggers, the message cjanges from
`<t> resist the spell.`
to
`<t> resists the spell!`

I would comment out that line and leave a TODO comment statying that the following things need verification:
- The messages returned when Monberaux resists paralyze. (To know the correct modifier)
- If he is actually immune. (To know the needed value of said modifier OR if a completely different method [Immunity] is needed)

Xaver-DaRed commented on scripts/globals/spells/trust/monberaux.lua:
```diff
@@ -14,16 +14,38 @@ spellObject.onSpellCast = function(caster, target, spell)
+    -- TODO: Add Holy Water skill (removes curse, zombie, and doom)
+    -- TODO: Add Cover ability with proper conditions (stand behind Monberaux when you have top enmity)
+    -- TODO: Add mechanic for turning status removal skills into AoE -- https://www.bg-wiki.com/ffxi/BGWiki:Trusts#Monberaux
+    
```
You have whitespace in this line. Remove it so the CI can do it's job, please.

thecursedcometh2 committed:
Commented out para immunity and added todo. 

Also removed whitespace

thecursedcometh2 commented:
Add requested changes. And I'm sorry, I don't know why the whitespace keeps showing up. Out of curiosity, which text editor do you use for lua files? I've been using Notepad++

Xaver-DaRed commented:
> Add requested changes. And I'm sorry, I don't know why the whitespace keeps showing up. Out of curiosity, which text editor do you use for lua files? I've been using Notepad++

I sometimes use Notepad++ and simetimes I used VS code or even Visual studio 2019

In notepad++ there is an option to show whitespace. In the line I pointed to, you will be able to delete spaces before actually deleting the line

Xaver-DaRed commented:
Also, while we are here, what do you use to push changes?

thecursedcometh2 commented:
> Also, while we are here, what do you use to push changes to github?

I'll definitely turn on that option. When I am doing small files (singular ones like Monberaux.lua) I push directly through the website. But when I am working on larger amounts of files, I use Github Desktop.

Xaver-DaRed commented:
I would recommend you to look up Gitkraken. It's a great tool, similar to Github Desktop, but actually good and easy to use once you get used to it.

Also, maybe familiarize with what "amend" and "Rebase" are and how they are used. For future PRs they will come in handy

thecursedcometh2 commented:
> I would recommend you to look up Gitkraken. It's a great tool, similar to Github Desktop, but actually good and easy to use once you get used to it.
> 
> Also, maybe familiarize with what "amend" and "Rebase" are and how they are used. For future PRs they will come in handy

Ah, I see. Gitkraken does look useful. I had also been looking for a way to do what rebase does, but I couldn't find anything so I ended up cherry picking commits instead (for my other projects, not this one). Thanks for the tips!

TeoTwawki commented:
> Added the changes that Zach requested. I also added "mob:addMod(xi.mod.PARALYZERES, 100)" because on retail he is functionally immune to paralyze. However, I don't know for sure if 100 = paralyze immune. Could anyone verify this? Not sure where to find mod values.

immunity and resistance are diff things. we usually do immunities in mob_pools.sql rather than in script.

TeoTwawki commented:
> > Also, while we are here, what do you use to push changes to github?
> 
> I'll definitely turn on that option. When I am doing small files (singular ones like Monberaux.lua) I push directly through the website. But when I am working on larger amounts of files, I use Github Desktop.

Avoid github desktop and also avoid using the website interface for adding and updating - those 2 systems should be avoided like the plague they are. they just lead to unexplained difficulties for you that you don't know how to explain or fix. Btw most editors including notepad++ have settings to show the whitespace visibly, and I recommend getting editorconfig to automate most of that for you (there is a notepad++ extension for it).

thecursedcometh2 commented:
> > Added the changes that Zach requested. I also added "mob:addMod(xi.mod.PARALYZERES, 100)" because on retail he is functionally immune to paralyze. However, I don't know for sure if 100 = paralyze immune. Could anyone verify this? Not sure where to find mod values.
> 
> immunity and resistance are diff things. we usually do immunities in mob_pools.sql rather than in script.

I went with resistance because apparently on retail he can have the paralysis status, but it will never proc. So I wasn't sure whether or not to just make him immune. But I will put mods in the sql from now on.

TeoTwawki commented:
> > > Added the changes that Zach requested. I also added "mob:addMod(xi.mod.PARALYZERES, 100)" because on retail he is functionally immune to paralyze. However, I don't know for sure if 100 = paralyze immune. Could anyone verify this? Not sure where to find mod values.
> > 
> > 
> > immunity and resistance are diff things. we usually do immunities in mob_pools.sql rather than in script.
> 
> I went with resistance because apparently on retail he can have the paralysis status, but it will never proc. So I wasn't sure whether or not to just make him immune. But I will put mods in the sql from now on.

I'm just explaining the difference on application - immunities aren't mods, its a value baked into the actual entity which is why its being loaded from sql. You really only want immunity if the text in game says "completely resists" meaning it can NEVER land no matter how much macc you have. Wikis will often say "immune" when its just "well I couldn't land one!" by some editor could couldn't overcome the resist rates. (for years ppl thought lesser sky NMs had all these immunities, till LV cap got raised and suddenly everyone had massive skill caps to work with)

WinterSolstice8 commented:
> I went with resistance because apparently on retail he can have the paralysis status, but it will never proc.

Then that means he doesn't need paralyze resistance; mobskills cannot be paralyzed. Go ahead and check:
`setmod paralyze 100` when targeting Monberaux.

thecursedcometh2 commented:
> > I went with resistance because apparently on retail he can have the paralysis status, but it will never proc.
> 
> Then that means he doesn't need paralyze resistance; mobskills cannot be paralyzed. Go ahead and check: `setmod paralyze 100` when targeting Monberaux.

Oh yeah, you're right. Huh, didn't know it worked like that. Looks like I can remove the paralyze todo then.

thecursedcometh2 committed:
Removed paralyze resist todo

TeoTwawki commented:
and this here is why I coined the phrase "shit wiki says"

thecursedcometh2 commented:
> and this here is why I coined the phrase "shit wiki says"

True lol. It's just kinda hard to tell on retail where you can't see the mods and stats of an entity.
